src/sentry/models/groupinbox.py

```diff
@@ -35,6 +50,17 @@ class Meta:
 
 
 def add_group_to_inbox(group, reason, reason_details=None):
+    from sentry.snuba.query_subscription_consumer import InvalidSchemaError
+
+    if reason_details is not None:
+        if "until" in reason_details and reason_details["until"] is not None:
+            reason_details["until"] = reason_details["until"].replace(microsecond=0).isoformat()
+
+    try:
+        jsonschema.validate(reason_details, INBOX_REASON_DETAILS)
+    except jsonschema.ValidationError:
+        raise InvalidSchemaError("GroupInbox reason_details does not match schema")
```

Comment:
It would be good to have the JSON schema error here.

Reply:
Do you think I should raise the jsonschema error instead? I was pattern matching against how Dan had done this in `query_subscription_consumer.py`

Reply:
`InvalidSchemaError` inherits from `InvalidMessageError`, and is basically specific to the subscription consumer. It might be fine to just let the `ValidationError` bubble up here

Reply:
One concern here... if this somehow fails in `post_process_group` it'd break processing for that group... Not sure what the best option is there.

We might be better off just not storing `reason_details` when there's a validation error and logging the error so that we can inspect it, but not break things.

Reply:
If only logging is used, will end users still see validation errors if they make invalid requests? I agree the post_process_group scenario shouldn't fail and we should log errors and skip saving there.

Reply:
@markstory I don't think this will be used by end users. The only situation the reason_details are used right now is when a group is unignored automatically (it basically stores a copy of GroupSnooze so we can display more details about why it was unignored and placed into the inbox on the frontend).

So I think just logging the error and not storing the details could work - given that this case shouldn't happen and it's likely a bug we need to fix if it does